corehq/apps/reports/standard/sms.py

```diff
@@ -201,6 +203,24 @@ def get_message_type_filter(self):
             return lambda message_types: len(filtered_types.intersection(message_types)) > 0
         return lambda message_types: True
 
+    def get_location_filter(self):
+        locations = []
+        location = AsyncLocationFilter.get_value(self.request, self.domain)
+        if location:
+            reporting_types = LocationType.objects.filter(domain=self.domain, administrative=False)
+            loc = SQLLocation.objects.get(location_id=location)
+            if loc.location_type in reporting_types:
+                locations.append(loc.location_id)
+
+            def get_reporting_locations_ids(parent_loc):
```

Comment:
also not blocking but i like to make it explicit with a comment when a function uses closures so that someone doesn't try to move it later.